test/e2e_node/podresources_test.go

```diff
@@ -821,6 +824,49 @@ var _ = SIGDescribe("POD Resources [Serial] [Feature:PodResources][NodeFeature:P
 			})
 		})
 	})
+
+	ginkgo.Context("when querying /metrics", func() {
+		ginkgo.BeforeEach(func() {
+			// ensure APIs have been called at least once
+			endpoint, err := util.LocalEndpoint(defaultPodResourcesPath, podresources.Socket)
+			framework.ExpectNoError(err)
+
+			cli, conn, err := podresources.GetV1Client(endpoint, defaultPodResourcesTimeout, defaultPodResourcesMaxSize)
+			framework.ExpectNoError(err)
+			defer conn.Close()
+
+			_, err = cli.List(context.TODO(), &kubeletpodresourcesv1.ListPodResourcesRequest{})
+			framework.ExpectNoError(err)
+
+			_, err = cli.GetAllocatableResources(context.TODO(), &kubeletpodresourcesv1.AllocatableResourcesRequest{})
+			framework.ExpectNoError(err)
+		})
+
+		ginkgo.It("should report the values for the podresources metrics", func() {
+			// we updated the kubelet config in BeforeEach, so we can assume we start fresh.
+			// being [Serial], we can also assume noone else but us is running pods.
+			ginkgo.By("Checking the value of the podresources metrics")
+
+			matchResourceMetrics := gstruct.MatchKeys(gstruct.IgnoreExtras, gstruct.Keys{
+				"kubelet_pod_resources_endpoint_requests_total": gstruct.MatchAllElements(nodeID, gstruct.Elements{
+					"": timelessSampleAtLeast(1),
+				}),
+				"kubelet_pod_resources_endpoint_requests_list": gstruct.MatchAllElements(nodeID, gstruct.Elements{
+					"": timelessSampleAtLeast(1),
+				}),
+				"kubelet_pod_resources_endpoint_requests_get_allocatable": gstruct.MatchAllElements(nodeID, gstruct.Elements{
+					"": timelessSampleAtLeast(1),
+				}),
+				// not checking errors: the calls don't have non-catastrophic (e.g. out of memory) error conditions yet.
+			})
+
+			ginkgo.By("Giving the Kubelet time to start up and produce metrics")
+			gomega.Eventually(getPodResourcesMetrics, 1*time.Minute, 15*time.Second).Should(matchResourceMetrics)
```

Comment:
consider adding `KubeletRestartTimout` to test/e2e/framework/timeouts.go

Reply:
I wonder if I should create a separate PR to do a sweeping change, because I took example from resource_metrics_test.go that can benefit from a similar change.

Reply:
yes, I noticed the same in other PR as well. Separate PR would be great

Reply:
sounds good to me.